lib/ui/painting/multi_frame_codec.cc

```diff
@@ -164,12 +177,13 @@ Dart_Handle MultiFrameCodec::getNextFrame(Dart_Handle callback_handle) {
 
   const auto& task_runners = dart_state->GetTaskRunners();
 
-  task_runners.GetIOTaskRunner()->PostTask(fml::MakeCopyable(
-      [callback = std::make_unique<DartPersistentValue>(
-           tonic::DartState::Current(), callback_handle),
-       this, trace_id, ui_task_runner = task_runners.GetUITaskRunner(),
-       io_manager = dart_state->GetIOManager()]() mutable {
-        GetNextFrameAndInvokeCallback(
+  task_runners.GetIOTaskRunner()->PostTask(
+      fml::MakeCopyable([callback = std::make_unique<DartPersistentValue>(
+                             tonic::DartState::Current(), callback_handle),
+                         state = state_, trace_id,
+                         ui_task_runner = task_runners.GetUITaskRunner(),
+                         io_manager = dart_state->GetIOManager()]() mutable {
+        state->GetNextFrameAndInvokeCallback(
```

Comment:
> you can do what you are doing in line 144.

Wait. Why are we even invoking the next frame callback on a muli-frame codec that has died (which would be indicated by the weak pointer lock failing)?

Instead of invoking the next frame callback when the weak pointer lock fails, scheduled a task onto the UI thread that just releases the callback but not invoke it (that has Dart state and you cant just drop it on the IO thread).

Reply:
I'm not clear on where it's safe to create the weak_ptr for the State.  I can't do it on the UI thread, and by the time I get to the IO thread the UI object might have been collected.

Reply:
Ohhh you mean a std::weak_ptr.  I get it now.

Reply:
For some reason I was thinking we had to still make the callback because some Dart code might be `await`ing the completion of the callback, but now that I think about it that doesn't make sense, since the only way this can happen is if the engine is collected.  If anyone is actually awaiting the callback, the codec must still be alive.  I'm correct in that thinking, right?

Reply:
Done - I believe

Reply:
Only question is whether we need to guard a bit more so that if we get collected _after_ this (e.g. we're still alive before we tell Skia to decode a frame, but we're dead after Skia finishes) - will this still be safe?

Reply:
The lock will fail in that case.